api/v1alpha1/common_types.go

```diff
@@ -71,6 +72,10 @@ var (
 
 type Selector string
 
+type ClusterSelector struct {
```

Comment:
Do you envision depracating the old Selector  few releases after we introduce new type? Or taking care of that on upgrade by moving to v1beta1 for instance?

Reply:
Yes, I don't know how you would like to handle crd lifecycles. For me its always new additions are okay on the same api version. Removal on the next api version bump. So i would suggest the second option

Reply:
Thank you. I have been thinking to move to v1beta1 for some time now (I think two years is more than enough to justify such change). I guess we can do while introducing this feature (and I will help of course). 

Saying that just so we only have one way to select a cluster (the new ClusterSelector you are introducing) while also not breaking existing deployment (handful of companies are using Sveltos in production).